pilot/pkg/proxy/envoy/v2/monitoring.go

```diff
@@ -102,6 +102,22 @@ var (
 	rdsSendErrPushes  = pushes.With(typeTag.Value("rds_senderr"))
 	rdsBuildErrPushes = pushes.With(typeTag.Value("rds_builderr"))
 
+	pushTime = monitoring.NewDistribution(
+		"pilot_xds_push_time",
+		"Total time Pilot takes to push lds, rds, cds and eds.",
+		[]float64{1, 3, 5, 10, 20, 30, 50, 100},
+		monitoring.WithLabels(typeTag),
+	)
+
+	cdsPushTime    = pushTime.With(typeTag.Value("cds"))
+	cdsPushErrTime = pushTime.With(typeTag.Value("cds_pusherr"))
```

Comment:
do we need this? not sure it's useful now that we have the per xds metrics

the original motivation was that if we failed out early our metrics would be lower but that's not really an issue now?

Reply:
No we don't need it. I missed a point here that we already report push failure count with `pilot_xds_pushes` metric. And we can query `pilot_xds_pushes` to know if a reported push time is for a succeeded or failed push. Let me remove them.